functorch/_src/eager_transforms.py

```diff
@@ -243,12 +243,16 @@ def vjp(func: Callable, *primals, has_aux=False):
             primals_out = func(*diff_primals)
 
             if has_aux:
+                # TODO: Add a test for that once vjp/jvp both have has_aux option
+                if not isinstance(primals_out, tuple):
+                    raise TypeError("Function output should be a tuple if has_aux is True")
```

Comment:
Should we assert it is a tuple of size 2?

Reply:
yes, we can do that. I'll update the code